### PR TITLE
Wireport

### LINKED ISSUE #168: WirePorts
Re-add "WirePorts".

Basically, they are very simple Components that are used to split wires and are rendered as a single port.

Example:
screen shot 2019-02-05 at 11 18 22 am

Try to write simple unit tests before implementing them to make sure they act appropriately.

Functionally, they act just like BUFGates.

For implementing them they'd be Component with exactly 1 input and output.

For rendering, you'd probably want to just put an if-statement before the rendering of the IOPorts in the ComponentRenderer.ts checking if the object is a WirePort and if it is then have a specific rendering process for them and 'return' early.
Or, potentially a better solution would be to set the origin/target of each input/output port to be at the origin of the local transform and the rendering of the port should just automatically happen but it could cause some rendering issues.

### PR DESCRIPTION
**Pull request by nprisament**
Adds wireports to the refactor

does not implement inserting wireports, deleting wireports, or reassigning wireports.

Fix #168

### CODE REVIEW

File: tests/public/ts/models/ioobjects/WirePort.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import "jest";
+
+import {CircuitDesigner} from "../../../../../site/public/ts/models/CircuitDesigner";
+import {WirePort}   from "../../../../../site/public/ts/models/ioobjects/WirePort";
+import {Switch}   from "../../../../../site/public/ts/models/ioobjects/inputs/Switch";
+import {LED}             from "../../../../../site/public/ts/models/ioobjects/outputs/LED";
+
+describe("Constant Low", () => {
```
Comment (LeonMontealegre) [inline tests/public/ts/models/ioobjects/WirePort.test.ts]: Describe should describe the component you're testing

File: site/public/ts/models/ioobjects/WirePort.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	public getDisplayName() {
+		return "Port";
+	}
+
+	public getImageName() {
+		return "";
+	}
+
+    public getXMLName(): string {
```
Comment (LeonMontealegre) [inline site/public/ts/models/ioobjects/WirePort.ts]: Spacing is weird here

File: site/public/ts/controllers/MainDesignerController.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ import {ANDGate}  from "../models/ioobjects/gates/ANDGate";
 import {ORGate}  from "../models/ioobjects/gates/ORGate";
```
Comment (LeonMontealegre) [inline site/public/ts/controllers/MainDesignerController.ts]: Remove changes to MainDesignerController before pushing please 🙏

File: tests/public/ts/models/ioobjects/WirePort.test.ts (hunk 1 above)
Comment (nprisament) [reply tests/public/ts/models/ioobjects/WirePort.test.ts]: fixed

File: site/public/ts/models/ioobjects/WirePort.ts (hunk 2 above)
Comment (nprisament) [reply site/public/ts/models/ioobjects/WirePort.ts]: fixed, for some reason, the tabs don't seem to be 4 spaces in this github viewer

File: site/public/ts/controllers/MainDesignerController.ts (hunk 3 above)
Comment (nprisament) [reply site/public/ts/controllers/MainDesignerController.ts]: done

**Pull request merged**